internal/cmd/cluster/cluster.go

```diff
@@ -25,68 +26,71 @@ var subcommands = []*cli.Command{
 	// Discover(),
 }
 
+var flags = []cli.Flag{
+	&cli.StringSliceFlag{
+		Name:    "addr",
+		Aliases: []string{"a"},
+		Usage:   "static bootstrap `ADDR`",
+		EnvVars: []string{"WW_ADDR"},
+	},
+	&cli.StringFlag{
+		Name:    "discover",
+		Aliases: []string{"d"},
+		Usage:   "bootstrap discovery `ADDR`",
+		Value:   "/ip4/228.8.8.8/udp/8822/multicast/lo0",
+		EnvVars: []string{"WW_DISCOVER"},
+	},
+	&cli.StringFlag{
+		Name:    "ns",
+		Usage:   "cluster namespace",
+		Value:   "ww",
+		EnvVars: []string{"WW_NS"},
+	},
+	&cli.DurationFlag{
+		Name:    "timeout",
+		Usage:   "dial timeout",
+		Value:   time.Second * 15,
+		EnvVars: []string{"WW_CLIENT_TIMEOUT"},
+	},
+}
+
 func Command() *cli.Command {
 	return &cli.Command{
-		Name:    "cluster",
-		Usage:   "cli client for wetware clusters",
-		Aliases: []string{"client"}, // TODO(soon):  deprecate
-		Flags: []cli.Flag{
-			&cli.StringSliceFlag{
-				Name:    "addr",
-				Aliases: []string{"a"},
-				Usage:   "static bootstrap `ADDR`",
-				EnvVars: []string{"WW_ADDR"},
-			},
-			&cli.StringFlag{
-				Name:    "discover",
-				Aliases: []string{"d"},
-				Usage:   "bootstrap discovery `ADDR`",
-				Value:   "/ip4/228.8.8.8/udp/8822/multicast/lo0",
-				EnvVars: []string{"WW_DISCOVER"},
-			},
-			&cli.StringFlag{
-				Name:    "ns",
-				Usage:   "cluster namespace",
-				Value:   "ww",
-				EnvVars: []string{"WW_NS"},
-			},
-			&cli.DurationFlag{
-				Name:    "timeout",
-				Usage:   "dial timeout",
-				Value:   time.Second * 15,
-				EnvVars: []string{"WW_CLIENT_TIMEOUT"},
-			},
-		},
+		Name:        "cluster",
+		Usage:       "cli client for wetware clusters",
+		Aliases:     []string{"client"}, // TODO(soon):  deprecate
+		Flags:       flags,
 		Subcommands: subcommands,
 	}
 }
 
 func setup() cli.BeforeFunc {
-	return func(c *cli.Context) error {
+	return func(c *cli.Context) (err error) {
 		app = fx.New(
 			runtime.Prelude(runtimeutil.New(c)),
 			fx.StartTimeout(c.Duration("timeout")),
-			fx.Populate(&logger, &node),
+			fx.Populate(&logger, &dialer),
 			runtime.Client())
 
 		ctx, cancel := context.WithTimeout(
 			c.Context,
 			app.StartTimeout())
 		defer cancel()
 
+		if node, err = dialer.Dial(ctx); err != nil {
```

Comment:
My previous (incorrect) comment notwithstanding, we need to add the timeout.  This is a simple matter of replacing `app.StartTimeout()` with `c.Duration("timeout")`.